### PR TITLE
Fix issues around Database permissions

### PR DESCRIPTION
**Pull request by mistercrunch**

### CODE REVIEW

File: superset/security.py [hunk 1]
```diff
@@ -410,8 +412,12 @@ def set_perm(self, mapper, connection, target):  # noqa
                 .values(perm=target.get_perm()),
             )
 
-        # add to view menu if not already exists
         permission_name = 'datasource_access'
+        from superset.models.core import Database
+        if mapper.class_ == Database:
```
Comment (mistercrunch) [inline superset/security.py]: FYI , `superset init` would create the proper `database_access` (one for each database), but the code here (executed on update/insert) would create/update a `datasource_access` entry for the database

File: superset/views/core.py [hunk 2]
```diff
@@ -109,18 +109,27 @@ def is_owner(obj, user):
...
 
         # TODO(bogdan): add `schema_access` support here
         datasource_perms = self.get_view_menus('datasource_access')
+        database_perms = self.get_view_menus('database_access')
         query = (
             query.outerjoin(SQLTable, self.model.datasource_id == SQLTable.c.id)
             .outerjoin(models.Database, models.Database.id == SQLTable.c.database_id)
             .filter(or_(
-                models.Database.perm.in_(datasource_perms),
+                models.Database.perm.in_(database_perms),
```
Comment (mistercrunch) [inline superset/views/core.py]: This appears to be using the wrong entry.

**Review by graceguo-supercat — APPROVED** [review]
LGTM

File: superset/views/core.py [hunk 3]
```diff
@@ -285,6 +294,7 @@ class DatabaseView(SupersetModelView, DeleteMixin, YamlExportMixin):  # noqa
         'allow_multi_schema_metadata_fetch': _('Allow Multi Schema Metadata Fetch'),
         'backend': _('Backend'),
     }
+    base_filters = [['id', DatabaseFilter, lambda: []]]
```
Comment (betodealmeida) [inline superset/views/core.py]: I think this broke stuff on our end, users can see DBs in SQL Lab. I commented it out and permissions work correctly.

Comment (mistercrunch) [reply superset/views/core.py]: Do you mean "can't see" (you wrote "can see").  This may require adding `all_database_access` to `alpha`

Comment (mistercrunch) [reply superset/views/core.py]: https://github.com/apache/incubator-superset/pull/7271

**Pull request merged**